Transformers/Java/src/test/java/com/github/ciselab/lampion/transformations/RenameVariableTransformerTests.java

```diff
@@ -0,0 +1,276 @@
+        CtMethod methodB = (CtMethod) ast.filterChildren(c -> c instanceof CtMethod).list().get(1);
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        TransformationResult result = new EmptyTransformationResult();
+        for(int i = 0; i<5;i++)
+            result = transformer.applyAtRandom(ast);
+
+        // Check if both methods still have both parameters as initially set
+        // The "," and ")" are important,
+        // as otherwise the test becomes flaky if the random variable name starts with a or b
+        List<CtVariable> variables = methodA.filterChildren(c -> c instanceof CtLocalVariable).list();
+        variables.addAll(methodB.filterChildren(c -> c instanceof CtLocalVariable).list());
+        Predicate<CtMethod> isAltered = m -> variables.stream().allMatch(n -> !n.getSimpleName().equals("a") || !n.getSimpleName().equals("b"));
+
+        boolean methodAAltered = isAltered.test(methodA);
+        boolean methodBAltered = isAltered.test(methodB);
+
+
+        // The operator "^" is the XOR operator
+        assertTrue(methodAAltered && methodBAltered);
+        assertEquals(new EmptyTransformationResult(),result);
+    }
+
+    @Test
+    void oneParameterOneVariable() {
+        CtElement ast = addOneLocalVariableExample();
+        CtMethod astMethod = (CtMethod) ast.filterChildren(c -> c instanceof CtMethod).list().get(0);
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        List<CtVariable> variables = astMethod.filterChildren(c -> c instanceof CtLocalVariable).list();
+        assertEquals(1, variables.size());
+    }
+
+    @Tag("Regression")
+    @Test
+    void amountOfParametersTest() {
+        CtElement ast = differentDeclarationExample();
+        CtMethod astMethod = (CtMethod) ast.filterChildren(c -> c instanceof CtMethod).list().get(0);
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        List<CtVariable> before = astMethod.filterChildren(c -> c instanceof CtLocalVariable).list();
+        assertEquals(2, before.size());
+
+        for(int i = 0; i<20;i++) {
+            transformer.applyAtRandom(astMethod);
+        }
+        List<CtVariable> after = astMethod.filterChildren(c -> c instanceof CtLocalVariable).list();
+        assertEquals(2, after.size());
+    }
+
+    @Test
+    void applyWithFullRandom_shouldGiveNonEmptyResult(){
+        CtElement ast = addBothExample();
+
+        RenameVariableTransformer transformer = new RenameVariableTransformer();
+
+        transformer.setFullRandomStrings(true);
+
+        TransformationResult result = transformer.applyAtRandom(ast);
+
+        assertNotEquals(new EmptyTransformationResult(),result);
+    }
+
+    @Test
+    void applyToMethodWithoutParametersOrVariables_returnsEmptyTransformationResult(){
+        CtClass ast = Launcher.parseClass("package lampion.test; class A { " +
```

Comment:
modern java can do block-strings, that might look nicer

Reply:
For this, the maven release should be changed to 15 instead of 14, which it is right now. I'll push the change for which this is done. If we don't want this it should be reverted.